lib/default.nix

```diff
@@ -27,6 +27,24 @@ _:
     }:
       assert lib.assertMsg (!(isAzure && sgx_default_qcnl_conf != null)) "sgx_default_qcnl_conf can't be set for Azure";
       let
+        manifestRecursiveMerge = base: mod: lib.recursiveUpdate base (
+          mod // {
+            # manually merge the relevant arrays / strings
+            fs.mounts =
+              if lib.attrsets.hasAttrByPath [ "fs" "mounts" ] mod
+              then base.fs.mounts ++ mod.fs.mounts
+              else base.fs.mounts;
+            sgx.trusted_files =
+              if lib.attrsets.hasAttrByPath [ "sgx" "trusted_files" ] mod
+              then base.sgx.trusted_files ++ mod.sgx.trusted_files
+              else base.sgx.trusted_files;
+            loader.env.LD_LIBRARY_PATH =
+              if lib.attrsets.hasAttrByPath [ "loader" "env" "LD_LIBRARY_PATH" ] mod
+              then base.loader.env.LD_LIBRARY_PATH + ":" + mod.loader.env.LD_LIBRARY_PATH
+              else base.loader.env.LD_LIBRARY_PATH;
+          }
```

Comment:
I'm not proficient in Nix, but I asked ChatGPT if it could make it more concise. Here is the result:

```nix
let
  mergeAttr = path: attr: base: mod: lib.attrsets.hasAttrByPath path mod ? attr base ++ attr mod : attr base;
  manifestRecursiveMerge = base: mod: lib.recursiveUpdate base (
    mod // {
      fs.mounts = mergeAttr ["fs" "mounts"] (b: b.fs.mounts) base mod;
      sgx.trusted_files = mergeAttr ["sgx" "trusted_files"] (b: b.sgx.trusted_files) base mod;
      loader.env.LD_LIBRARY_PATH =
        lib.attrsets.hasAttrByPath ["loader" "env" "LD_LIBRARY_PATH"] mod
        ? base.loader.env.LD_LIBRARY_PATH + ":" + mod.loader.env.LD_LIBRARY_PATH
        : base.loader.env.LD_LIBRARY_PATH;
    }
  );
```
(not sure if that works)

Anyway, good enough for me. ✅

Reply:
nice!

Reply:
```
❯ nix repl
Welcome to Nix 2.18.2. Type :? for help.

nix-repl> if true then 1 else 0 
1

nix-repl> true ? 1 : 0
error: syntax error, unexpected INT, expecting ID or OR_KW or DOLLAR_CURLY or '"'

       at «string»:1:8:

            1| true ? 1
             |        ^

```

Reply:
I can offer this, if it makes you happy?
```nix
with lib.attrsets; let
        mergeAttr = attrPath: merge: base: mod: if hasAttrByPath attrPath mod then merge (base // mod) mod else base // mod;
        mergeArrayByPath = path: base: mod: if hasAttrByPath path mod then (getAttrFromPath path base) ++ (getAttrFromPath path mod) else (getAttrFromPath path base);
        mergeEnvPathByPath = path: base: mod: if hasAttrByPath path mod then (getAttrFromPath path base) + ":" + (getAttrFromPath path mod) else (getAttrFromPath path base);
        manifestRecursiveMerge = base: mod: lib.recursiveUpdate base (
          mod // {
            fs.mounts = mergeArrayByPath [ "fs" "mounts" ] base mod;
            sgx.trusted_files = mergeArrayByPath [ "sgx" "trusted_files" ] base mod;
            loader.env.LD_LIBRARY_PATH = mergeEnvPathByPath [ "loader" "env" "LD_LIBRARY_PATH" ] base mod;
          });
        in  manifestRecursiveMerge
```

Reply:
yeah, I am pretty happy with this.